firmware/powerboard2/code/app/ADC.c

```diff
@@ -0,0 +1,149 @@
+/*
+ * ADC.c
+ */
+
+#include "ADC.h"
+
+/* INCLUDES */
+#include "stm32f4xx.h"
+#include <stdbool.h>
+
+/* DEFINES */
+
+/* TYPEDEFS */
+typedef struct
+{
+    uint16_t rawData[ADC_CHANNEL_COUNT];
+} ADC_data_S;
+
+/* PRIVATE DATA DECLARAION */
+static ADC_data_S ADC_data;
+
+static const uint32_t ADC_softwareChannelToADCChannelMapping[ADC_CHANNEL_COUNT] =
+{
+    [ADC_CHANNEL_ADC_BAT0_1]          = ADC_Channel_15,
+    [ADC_CHANNEL_ADC_BAT0_2]          = ADC_Channel_14,
+    [ADC_CHANNEL_ADC_BAT0_3]          = ADC_Channel_7,
+    [ADC_CHANNEL_ADC_BAT0_4]          = ADC_Channel_6,
+    [ADC_CHANNEL_ADC_BAT0_5]          = ADC_Channel_5,
+    [ADC_CHANNEL_ADC_BAT0_6]          = ADC_Channel_4,
+	[ADC_CHANNEL_ADC_BAT1_1]          = ADC_Channel_3,
+	[ADC_CHANNEL_ADC_BAT1_2]          = ADC_Channel_2,
+    [ADC_CHANNEL_ADC_BAT1_3]          = ADC_Channel_1,
+    [ADC_CHANNEL_ADC_BAT1_4]          = ADC_Channel_0,
+    [ADC_CHANNEL_ADC_BAT1_5]          = ADC_Channel_13,
+	[ADC_CHANNEL_ADC_BAT1_6]          = ADC_Channel_12,
+	[ADC_CHANNEL_ADC_EXT_PRESSURE]    = ADC_Channel_10,
+	[ADC_CHANNEL_ADC_WATER_SENSE]     = ADC_Channel_11,
+};
+
+/* PRIVATE FUNCTION DECLARATION */
+static void ADC_private_initGPIOs(void);
```

Comment:
These declarations are not needed

Reply:
We should have this. I just didn't turn the warning on for this, but it's good practice. Some linters will complain.